lib/commonBindings.js

```diff
@@ -51,12 +53,68 @@ function guessType(attribute, device) {
     }
 
     if (attribute === constants.TIMESTAMP_ATTRIBUTE) {
-        return constants.TIMESTAMP_TYPE;
+        if (iotAgentLib.configModule.checkNgsi2()) {
+            return constants.TIMESTAMP_TYPE_NGSI2;
+        }
+        else {
+            return constants.TIMESTAMP_TYPE;
+        }
     } else {
         return constants.DEFAULT_ATTRIBUTE_TYPE;
     }
 }
 
+/**
+ * Converts the value to the corresponding type given by attribute parameter.
+ * null otherwise.
+ *
+ * @param {String} attribute        Name of the attribute to find.
+ * @param {String} value            Value of the attribute to convert.
+ * @param {Object} device           Device object containing all the information about a device.
+ * @return {Object}                 Attribute's value represented in the appropriate format.
+ */
+function convertValue(attribute, value, device) {
+
+    if (!iotAgentLib.configModule.checkNgsi2()) {
+        return value;
+    }
+
+    if (attribute === constants.TIMESTAMP_ATTRIBUTE) {
+        return value;
+    }
+
+    for (var i = 0; i < device.active.length; i++) {
+        if (device.active[i].name === attribute) {
+            var type = device.active[i].type;
+
+            if (type === 'Integer') {
+                return Number.parseInt(value);
+            }
+            else if (type === 'Float') {
+                return Number.parseFloat(value);
+            }
+            else if (type === 'Boolean') {
+                return (value === 'true' || value === '1');
+            }
+            else if (type === 'Null') {
```

Comment:
Looking to a similar code in the iotanode-lib-agent (https://github.com/telefonicaid/iotagent-node-lib/pull/556/files#diff-226f40312563bee0db028aec5e514ba0R178) this should be 'None' (I gess it impacts also in the unit test, all the "Null" cases should be changed to "None").

What do you think?

Reply:
Moreover, "None" is more aligned with NGSIv2 with regards to attributes with null value. See "Partial Representations" in http://telefonicaid.github.io/fiware-orion/api/v2/stable/:

> If value is null, then `None` is used.

Reply:
Fixed